src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -45,7 +47,7 @@ public ErrorLogger(Stream stream, string toolName, string toolFileVersion, Versi
 
         private void WriteToolInfo(string name, string fileVersion, Version assemblyVersion)
         {
-            _writer.WriteObjectStart("toolInfo");
+            _writer.WriteObjectStart("tool");
             _writer.Write("name", name);
             _writer.Write("version", assemblyVersion.ToString(fieldCount: 3));
```

Comment:
you should consider emitting a marker in the sarif file that provides an explicit version for your SARIF producer. in the SDK, we dump a custom property SarifLoggerVersion on the tools property. this isn't necessary, though, if other version information sufficiently identifies this specific point-in-time (perhaps your assembly file version suffices)?

@lgolding, our tolerance for additions to SARIF is very low. you could imagine, though, extending the 'tool' metadata to include a 'sarifLoggerVersion' for tools that integrate some other component for producing SARIF (that may be versioned separately from the root tool itself). i.e., promote the SDK custom property.